tnt-service/src/main/java/org/dcsa/tnt/service/EventSubscriptionService.java

```diff
@@ -0,0 +1,156 @@
+      .orElseThrow(() -> ConcreteRequestErrorMessageException.notFound("No event-subscription found with id = " + subscriptionID));
+
+    updateList(subscriptionID, eventSubscription.getEventTypes(), eventSubscriptionMapper::toDAO, EventSubscriptionEventType::new, eventSubscriptionRepository::deleteEventType, original.getEventTypes());
+    updateList(subscriptionID, eventSubscription.getTransportEventTypeCodes(), eventSubscriptionMapper::toDAO, EventSubscriptionTransportEventTypeCode::new, eventSubscriptionRepository::deleteTransportEventTypeCode, original.getTransportEventTypeCodes());
+    updateList(subscriptionID, eventSubscription.getShipmentEventTypeCodes(), eventSubscriptionMapper::toDAO, EventSubscriptionShipmentEventTypeCode::new, eventSubscriptionRepository::deleteShipmentEventTypeCode, original.getShipmentEventTypeCodes());
+    updateList(subscriptionID, eventSubscription.getEquipmentEventTypeCodes(), eventSubscriptionMapper::toDAO, EventSubscriptionEquipmentEventTypeCode::new, eventSubscriptionRepository::deleteEquipmentEventTypeCode, original.getEquipmentEventTypeCodes());
+    updateList(subscriptionID, eventSubscription.getDocumentTypeCodes(), eventSubscriptionMapper::toDAO, EventSubscriptionDocumentTypeCode::new, eventSubscriptionRepository::deleteDocumentTypeCode, original.getDocumentTypeCodes());
+
+    eventSubscriptionRepository.save(
+      original.toBuilder()
+        .callbackUrl(eventSubscription.getCallbackUrl())
+        .documentReference(eventSubscription.getDocumentReference())
+        .equipmentReference(eventSubscription.getEquipmentReference())
+        .transportCallReference(eventSubscription.getTransportCallReference())
+        .vesselIMONumber(eventSubscription.getVesselIMONumber())
+        .carrierExportVoyageNumber(eventSubscription.getCarrierExportVoyageNumber())
+        .universalExportVoyageReference(eventSubscription.getUniversalExportVoyageReference())
+        .carrierServiceCode(eventSubscription.getCarrierServiceCode())
+        .universalServiceReference(eventSubscription.getUniversalServiceReference())
+        .UNLocationCode(eventSubscription.getUNLocationCode())
+        .build()
+    );
+  }
+
+  @Transactional
+  public void updateSecret(UUID subscriptionID, EventSubscriptionSecretTO eventSubscriptionSecret) {
+    eventSubscriptionRepository.updateSecret(subscriptionID, eventSubscriptionSecret.secret());
+  }
+
+  private <S extends Enum<S>, T extends Enum<T>, TI extends EventSubscriptionEnumSetItem<T>> void setIfNotEmpty(
+    UUID subscriptionId,
+    Set<S> src,
+    Function<S, T> enumMapper,
+    BiFunction<UUID, T, TI> entityFactory,
+    Consumer<Set<TI>> target
+  ) {
+    if (src != null) {
+      target.accept(src.stream()
+        .map(enumMapper)
+        .map(e -> entityFactory.apply(subscriptionId, e))
+        .collect(Collectors.toSet()));
+    }
+  }
+
+  private <S extends Enum<S>, T extends Enum<T>, TI extends EventSubscriptionEnumSetItem<T>> void updateList(
```

Comment:
method is named updateList but it updates a Set

Reply:
Fixed